<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SPI_Master_GUI</class>
 <widget class="QWidget" name="SPI_Master_GUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>FT2232H SPI Master - Arduino Slave 통신</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="conn_group">
     <property name="title">
      <string>FT2232H 연결 설정</string>
     </property>
     <layout class="QGridLayout" name="conn_layout">
      <item row="0" column="0">
       <widget class="QLabel" name="url_label">
        <property name="text">
         <string>FTDI URL:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QLineEdit" name="url_edit">
        <property name="text">
         <string>ftdi://ftdi:2232h/1</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="cs_label">
        <property name="text">
         <string>CS:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="cs_spin">
        <property name="maximum">
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="freq_label">
        <property name="text">
         <string>주파수(Hz):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="freq_edit">
        <property name="text">
         <string>100000</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="mode_label">
        <property name="text">
         <string>SPI 모드:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="mode_combo">
        <item>
         <property name="text">
          <string>0</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>3</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="connect_btn">
        <property name="text">
         <string>연결</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="disconnect_btn">
        <property name="text">
         <string>연결 해제</string>
        </property>
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="spi_group">
     <property name="title">
      <string>SPI 통신 (RW_BIT + Address)</string>
     </property>
     <layout class="QGridLayout" name="spi_layout">
      <item row="0" column="0">
       <widget class="QLabel" name="addr_label">
        <property name="text">
         <string>주소 (0-127):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="addr_edit">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>01</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="data_label">
        <property name="text">
         <string>데이터 (Hex):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLineEdit" name="data_edit">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>5678</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="read_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>읽기 (RW=1)</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #4CAF50; color: white;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="write_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>쓰기 (RW=0)</string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #2196F3; color: white;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="test_label">
        <property name="text">
         <string>빠른 테스트:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="test_read1_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>읽기 Addr 0x01</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="test_read2_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>읽기 Addr 0x02</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="test_read3_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>읽기 Addr 0x03</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="test_write1_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>쓰기 Addr 0x01</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="QCheckBox" name="auto_checkbox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>자동 millis() 읽기 (Addr 0x03)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="resp_group">
     <property name="title">
      <string>마지막 응답</string>
     </property>
     <layout class="QHBoxLayout" name="resp_layout">
      <item>
       <widget class="QLabel" name="resp_addr_title">
        <property name="text">
         <string>주소:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="resp_addr_label">
        <property name="text">
         <string>--</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="resp_data_title">
        <property name="text">
         <string>데이터:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="resp_data_label">
        <property name="text">
         <string>--</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="resp_dec_title">
        <property name="text">
         <string>십진수:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="resp_dec_label">
        <property name="text">
         <string>--</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="log_group">
     <property name="title">
      <string>로그</string>
     </property>
     <layout class="QVBoxLayout" name="log_layout">
      <item>
       <widget class="QTextEdit" name="log_text">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>200</height>
         </size>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clear_log_btn">
        <property name="text">
         <string>로그 지우기</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>